src/common_interface/integrator_utils.jl

```diff
@@ -115,3 +115,17 @@ end
 @inline function DiffEqBase.get_du!(out,integrator::IDAIntegrator)
   out .= reshape(integrator.du,integrator.sizedu)
 end
+
+@inline function DiffEqBase.change_t_via_interpolation!(integrator::AbstractSundialsIntegrator,t)
+    integrator.t = t
+    integrator(integrator.u,integrator.t)
```

Comment:
this doesn't actually change `t` though? You need to `reinit!` the integrator since the Nordsieck vectors are now incorrect.

Reply:
I see, I did that here. https://github.com/JuliaDiffEq/Sundials.jl/blob/master/src/common_interface/integrator_utils.jl#L28 so that it happens after all of the discontinuity handling on the correct `u`.